main.go

```diff
@@ -3,94 +3,260 @@ package main
 	"github.com/grafana/detect-angular-dashboards/output"
 )
 
-const (
-	envGrafanaToken = "GRAFANA_TOKEN"
-)
-
-func newLogger(verboseFlag, jsonOutputFlag bool) *logger.LeveledLogger {
-	log := logger.NewLeveledLogger(verboseFlag)
-	if jsonOutputFlag {
-		// Redirect everything to stderr to avoid mixing with json output
-		log.Logger.SetOutput(os.Stderr)
-		log.WarnLogger.SetOutput(os.Stderr)
-	}
-	return log
-}
+const envGrafana = "GRAFANA_TOKEN"
 
-func getToken() (string, error) {
-	token := os.Getenv(envGrafanaToken)
-	if token == "" {
-		return "", fmt.Errorf("missing env var %q", envGrafanaToken)
-	}
-	return token, nil
+type Output struct {
+	mu   sync.Mutex
+	data []output.Dashboard
 }
 
 func main() {
-	versionFlag := flag.Bool("version", false, "print version number")
-	verboseFlag := flag.Bool("v", false, "verbose output")
-	jsonOutputFlag := flag.Bool("j", false, "json output")
-	skipTLSFlag := flag.Bool("insecure", false, "skip TLS verification")
-	flag.Parse()
+	f := flags.Parse()
 
-	if *versionFlag {
+	if f.Version {
 		fmt.Printf("%s %s (%s)\n", os.Args[0], build.LinkerVersion, build.LinkerCommitSHA)
 		os.Exit(0)
 	}
-	log := newLogger(*verboseFlag, *jsonOutputFlag)
+	log := newLogger(f.Verbose, f.JSONOutput)
 
 	token, err := getToken()
 	if err != nil {
-		_, _ = fmt.Fprintln(os.Stderr, err.Error())
+		log.Errorf("Failed to retrieve Grafana token: %s\n", err.Error())
+		os.Exit(1)
+	}
+	client := initializeClient(token, &f)
+
+	d := detector.NewDetector(log, client, gcom.NewAPIClient(), f.MaxConcurrency)
+
+	if f.Server != "" {
+		if err := runServerMode(&f, log, d); err != nil {
+			log.Errorf("%s\n", err)
+			os.Exit(1)
+		}
+		return
+	}
+
+	if err := runCLIMode(&f, log, d); err != nil {
+		log.Errorf("%s\n", err)
 		os.Exit(1)
 	}
+}
+
+// runServerMode runs the program in server (HTTP) mode.
+func runServerMode(flags *flags.Flags, log *logger.LeveledLogger, d *detector.Detector) error {
+	// Readiness flag using atomic boolean
+	var ready atomic.Bool
+	var once sync.Once
+
+	ticker := time.NewTicker(flags.Interval)
+	defer ticker.Stop()
+	log.Log("Running detection every %s", flags.Interval)
+
+	var out Output
+	go func() {
+		// Trigger for the first time
+		run := make(chan struct{}, 1)
+		run <- struct{}{}
+
+		for {
+			select {
+			case <-run:
+			case <-ticker.C:
+			}
+
+			// Run detection periodically
+			log.Log("Detecting Angular dashboards")
+			data, err := d.Run(context.Background())
+			if err != nil {
+				log.Errorf("%s\n", err)
+				continue
+			}
+
+			// Run detection periodically
+			log.Log("Updating Output Data")
+			out.mu.Lock()
+			out.data = data
+			out.mu.Unlock()
+
+			// Use sync.Once to set readiness only once
+			once.Do(func() {
+				ready.Store(true)
+				log.Log("Updating readiness probe to ready")
+			})
+		}
+	}()
+
+	http.HandleFunc("/output", func(w http.ResponseWriter, r *http.Request) {
+		handleOutputRequest(w, r, &out, log)
+	})
```

Comment:
Nit: After taking another look at this, I think it would make more sense to rename this handler to `/detections`. WDYT?

```suggestion
	http.HandleFunc("/detections", func(w http.ResponseWriter, r *http.Request) {
		handleDetectionsRequest(w, r, &out, log)
	})
```